### PR TITLE
fix: fix distributed tool

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

fix: fix distributed tool

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by zmsn-2077 — DISMISSED** [review]

File: omnisafe/utils/distributed_utils.py [hunk 1]
```diff
@@ -218,23 +241,20 @@ def mpi_avg_torch_tensor(value) -> None:
             raise NotImplementedError
 
 
-def mpi_statistics_scalar(value, with_min_and_max=False) -> tuple:
+def mpi_statistics_scalar(value: torch.Tensor, with_min_and_max=False) -> tuple:
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:244]:
```suggestion
def mpi_statistics_scalar(value: torch.Tensor, with_min_and_max: bool = False) -> Union[Tuple[np.ndarray, np.ndarray], Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]]:
```

File: omnisafe/utils/distributed_utils.py [hunk 2]
```diff
@@ -51,40 +59,39 @@ def mpi_avg_grads(module):
 
 
 def sync_params(module):
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:61]:
```suggestion
def sync_params(module: torch.nn.Module) -> None:
```

File: omnisafe/utils/distributed_utils.py [hunk 3]
```diff
@@ -115,36 +122,38 @@ def mpi_fork(parallel: int, bind_to_core=False, use_number_of_threads=False) ->
 
 
 def is_root_process() -> bool:
-    """is_root_process"""
+    """Judge whether the process is the root process."""
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:125]:
```suggestion
    """Test whether the process is the root process."""
```

**Review by XuehaiPan — CHANGES_REQUESTED** [review]
Regenerate TOC.

File: omnisafe/utils/distributed_utils.py [hunk 4]
```diff
@@ -115,36 +123,38 @@ def mpi_fork(parallel: int, bind_to_core=False, use_number_of_threads=False) ->
...
+def gather(*args, **kwargs) -> torch.Tensor:
+    """Gather operation."""
     return dist.gather(*args, **kwargs)
 
 
 def gather_and_stack(x_vector: np.ndarray) -> np.array:
     """Gather values from all tasks and return flattened list.
-    Note: only the root process owns valid data.
+
+    Input is a 1D array of size N, and output is a list of size N * MPI_world_size.
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:150]:
```suggestion
    Input is a 1D array of size ``N``, and output is a list of size ``N * MPI_world_size``.
```

File: omnisafe/utils/distributed_utils.py [hunk 5]
```diff
@@ -115,36 +123,38 @@ def mpi_fork(parallel: int, bind_to_core=False, use_number_of_threads=False) ->
...
 
-def gather(*args, **kwargs):
-    """gather"""
+def gather(*args, **kwargs) -> torch.Tensor:
+    """Gather operation."""
     return dist.gather(*args, **kwargs)
 
 
 def gather_and_stack(x_vector: np.ndarray) -> np.array:
     """Gather values from all tasks and return flattened list.
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:148]:
```suggestion
    """Gather values from all tasks and return a flattened list.
```

File: omnisafe/utils/distributed_utils.py [hunk 6]
```diff
@@ -41,50 +46,53 @@ def setup_torch_for_mpi():
...
 
-    Returns:
-        bool
-            True if process is parent process of MPI
+    .. note::
 
-    Usage:
-        if mpi_fork(n):  # forks the current script and calls MPI
-            sys.exit()   # exit single thread python process
+        Usage: if ``mpi_fork(n)`` : ``sys.exit()``
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:90]:
```suggestion
    Examples:
        .. code-block:: python

            if mpi_fork(n):  # forks the current script and calls MPI
                sys.exit()   # exit single thread python process
```

**Review by XuehaiPan — CHANGES_REQUESTED** [review]
Please fix the TOC in README.

File: omnisafe/utils/distributed_utils.py [hunk 7]
```diff
@@ -195,16 +215,20 @@ def mpi_op(value, operation):
...
 
-def mpi_avg_torch_tensor(value) -> None:
+def mpi_avg_torch_tensor(value: torch.Tensor) -> None:
     """Average a torch tensor over MPI processes.
-    Since torch and numpy share same memory space, tensors of dim > 0 can be
-    be manipulated through call by reference, scalars must be assigned.
 
+    Since torch and numpy share same memory space,
+    tensors of dim > 0 can be be manipulated through call by reference,
+    scalars must be assigned.
```
Comment (XuehaiPan) [inline omnisafe/utils/distributed_utils.py:228]: Wrap the docstrings at a specific column (e.g., at 100 characters per line).

**Pull request closed without merging**